### PR TITLE
Added tests for utils.py and config.py

### PR DESCRIPTION
**Pull request by nitinkgp23**
This PR adds tests for **utils.py**, **config.py** and **client.py**
- [x] Covers config.py to 100%
- [x] Covers utils.py to 81%

WIP for client.py:
- [x] SeriesInfo covered
- [ ] Present coverage of client.py : 26%

TODO:
- [ ] Find a way to write tests for older pandas version in utils.py

### CODE REVIEW

**Comment by kbg** [comment]
Your code does not work. You need to add the function arguments from the parametrize marker to the test function, e.g.

    @pytest.mark.parametrize('arg, exp', [...])
    def test_function(arg, exp):
       ...

Also, you made an error in the parametrize part of `test_pd_to_numeric_coerce()`: The arguments need to be a single string, i.e. `'arg, exp'`, not two separate strings.

Please test your code in the future and make sure that it runs through, before you make a pull request... :o)

**Comment by nitinkgp23** [comment]
Sorry, I was going to update the PR. I have corrected it now. :)

**Comment by kbg** [comment]
Thanks, it looks much better now. :)

About the tests for `to_datetime()`: You could re-use the time strings that were already used by `test_tai_string()` to write a sequence version of the test. This could be easily done by moving the arguments of the parametrize decorator to a global variable and then use this variable for the tests. For example:

```python
data_tai = [
    ('2010.05.01_TAI', pd.Timestamp('2010-05-01 00:00:00')),
    ('2010.05.01_00:00_TAI', pd.Timestamp('2010-05-01 00:00:00')),
    ('2010.05.01_00:00:00_TAI', pd.Timestamp('2010-05-01 00:00:00')),
    ('2010.05.01_01:23:45_TAI', pd.Timestamp('2010-05-01 01:23:45')),
    ('2013.12.21_23:32_TAI', pd.Timestamp('2013-12-21 23:32:00')),
    ('2013.12.21_23:32:34_TAI', pd.Timestamp('2013-12-21 23:32:34')),
    ]
data_tai_in = [s[0] for s in data_tai]
data_tai_out = pd.Series([s[1] for s in data_tai])


@pytest.mark.parametrize('time_string, expected', data_tai)
def test_tai_string(time_string, expected):
    assert drms.to_datetime(time_string) == expected


@pytest.mark.parametrize('sequence, expected', [
    (data_tai_in, data_tai_out),
    (np.array(data_tai_in), data_tai_out),
    (pd.Series(data_tai_in), data_tai_out),
    ])
def test_tai_sequence(sequence, expected):
    assert drms.to_datetime(sequence).equals(expected)
```

I think this would make it clearer to see, which cases are covered by the tests. In addition I think it would also be nice to add some tests that cover NaT results or empty input series.

I just realized that `drms.to_datetime()` does not support empty lists, which might cause some confusion when using this function with an empty list. The reason is that `pandas.Series([])` creates a series with a `float64` `dtype` by default, which does not have all the `.str.xxx()` members.

The easiest way to fix this (and possibly other issues) would probably be to add `dtype=str` to the Series conversion inside of `drms.to_datetime()`. I just opened a new bug report for this (#8). I'll try to write a quick fix tomorrow. If you would like to fix it, just go ahead and send another PR for it. :)

File: drms/tests/test_to_datetime.py [hunk 1]
```diff
@@ -52,6 +58,20 @@ def test_force_string(time_string, expected):
...
+    ])
+def test_time_series(time_series, expected):
+    drms.to_datetime(time_series).equals(expected)
+
+
+def test_corner_case():
+    assert type(drms.to_datetime('')) == pd._libs.tslib.NaTType
+    assert type(drms.to_datetime('foo')) == pd._libs.tslib.NaTType
+    assert type(drms.to_datetime([])) == pd.Series
+    assert drms.to_datetime([]).empty
```
Comment (nitinkgp23) [inline drms/tests/test_to_datetime.py]: This function needs to be improved. I was going to add a list of valid timestrings and invalid strings together, which would result in a Series with some NaT values. But I couldn't understand how to write `NaT`. The only thing that I was able to compare it with is its data type.

File: drms/tests/test_to_datetime.py [hunk 2]
```diff
@@ -52,6 +58,20 @@ def test_force_string(time_string, expected):
...
-# test_xx_list
-# test_xx_ndarray
-# test_xx_pandas_series
+@pytest.mark.parametrize('time_series, expected', [
+    (data_tai_in, data_tai_out),
+    (pd.Series(data_tai_in), data_tai_out),
+    (tuple(data_tai_in), data_tai_out),
+    (np.array(data_tai_in), np.array(data_tai_out)),
+    (dict(zip(tuple(range(len(data_tai_in))), tuple(data_tai_in))),
+     dict(zip(tuple(range(len(data_tai_out))), tuple(data_tai_out))))
```
Comment (nitinkgp23) [inline drms/tests/test_to_datetime.py]: Any better way to write this?

Comment (kbg) [inline drms/tests/test_to_datetime.py]: First of all: You forgot the `assert` keyword in the function body of `test_time_series()`. If you add the `assert` statement, you will see that the last two tests are failing.

You should always use `data_tai_out` as the expected value, because `drms.to_datetime()` always returns `pandas.Series` instances for non-scalar values. So instead of
```python
(np.array(data_tai_in), np.array(data_tai_out)),
```
you should write
```python
(np.array(data_tai_in), data_tai_out),
```
This test item should then succeed.

I don't think that we need the last test item. The fact that the `pandas.Series` constructor is called inside of `pandas.to_datetime()` is an internal implementation detail. So this might not work anymore in future implementations (and is not advertised anywhere in the documentation).

FYI:
```python
dict(enumerate(data_tai_in))
```
is equivalent to
```python
dict(zip(tuple(range(len(data_tai_in))), tuple(data_tai_in)))
```
But, as I said, better just remove this test.

File: drms/tests/test_to_datetime.py (hunk 1 above)
Comment (kbg) [inline drms/tests/test_to_datetime.py]: You can check if an object is an instance of a certain class by using the built-in `isinstance()` function. So instead of
```python
assert type(drms.to_datetime([])) == pd.Series
```
you should write
```python
assert isinstance(drms.to_datetime([]), pd.Series)
```

`pandas.NaT` values behave similar as `NaN` values. For example the expression `NaT != NaT` is always true. To check for `NaN` in NumPy, one usually uses `numpy.isnan()`. I think there does not exist a function in Pandas that can explicitly check for `NaT` (like `pandas.isnat()` or similar), but you can always use `pandas.isnull()` to check for invalid/missing values. This function would also return `True` for `NaN` or `None` values, but this should not be a problem in this case.

Alternatively you could also use `isinstance(x, type(pandas.NaT))` or `isinstance(x, pandas.NaTType)`, but I would just use `pandas.isnull()` instead.

File: drms/tests/test_to_datetime.py (hunk 2 above)
Comment (nitinkgp23) [inline drms/tests/test_to_datetime.py]: Ohh! I had tried doing both, and I was wondering why both the tests pass. Now, I see I missed the assert keyword :/

**Comment by nitinkgp23** [comment]
To make the tests pass, I had to rebase against my bugfix branch. So, there is a common commit "	Allow for empty input lists" in both the PRs. Will it be a problem when both of my branches get merged?

**Comment by kbg** [comment]
When the fix is merged into master, you can rebase this branch again and remove the redundant commit.

**Comment by wtbarnes** [comment]
Is there anything blocking this PR? This seems to just be adding and improving tests.

**Comment by nabobalis** [comment]
I have added these tests into #41.

**Comment by nabobalis** [comment]
Thanks for the extra tests @nitinkgp23
